runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -9405,20 +9481,20 @@ static TR::Register* inlineCompareAndSwapObjectNative(TR::Node* node, TR::CodeGe
 
       generateLabelInstruction(LABEL, node, begLabel, cg);
 
-      generateRegMemInstruction(CMPRegMem(use64BitClasses), node, tmp, generateX86MemoryReference(cg->getVMThreadRegister(), offsetof(J9VMThread, evacuateBase), cg), cg);
+      generateRegMemInstruction(CMP4RegMem, node, tmp, generateX86MemoryReference(cg->getVMThreadRegister(), offsetof(J9VMThread, evacuateBaseCompressed), cg), cg);
```

Comment:
On Z platforms CS runs on both 64 CR and nonCR, but not on 32. Same should be on X. But if you want to stage it and enable only CR now and later for nonCR that's ok, but nonCR should be supported eventually.

Reply:
Thank you for reminding. This version only supports 64 CR. There are more changes required beyond this PR to make CS on nonCR work. CS support on nonCR will be added after CR is fully enabled.